pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2059,6 +2137,10 @@ impl Timeline {
             Vec::new()
         };
 
+        if partial_compaction && !produced_delta_layers.is_empty() {
+            bail!("partial compaction cannot produce delta layers (for now)");
```

Comment:
WHy not assert? We're pretty liberal with asserts in the other parts of bottommost compaction code.

```suggestion
            bail!("implementation error: partial compaction should not be producing delta layers (for now)");
```